include/dl/dl_txt.h

```diff
@@ -61,16 +61,36 @@ dl_error_t DL_DLL_EXPORT dl_txt_pack_calc_size( dl_ctx_t dl_ctx, const char* txt
 		produced_bytes        - Number of bytes that would have been written to out_txt_instance if it was large enough.
 
 	Note:
-		Packed instance to unpack is required to be in current platform endian, if not DL_ERROR_ENDIAN_ERROR will be returned.
+		A stored packed instance to unpack is required to be in current platform endian, if not DL_ERROR_ENDIAN_ERROR will be returned.
+*/
+dl_error_t DL_DLL_EXPORT dl_txt_unpack( dl_ctx_t       dl_ctx,           dl_typeid_t type,
+                                        unsigned char* packed_instance,  size_t      packed_instance_size,
+                                        char*          out_txt_instance, size_t      out_txt_instance_size,
+                                        size_t*        produced_bytes );
+
+/*
+	Function: dl_txt_unpack_loaded
+	    Unpack a loaded binary (without header, with pointers) instance to text-format.
+
+	Parameters:
+	    dl_ctx                - Context to use.
+	    type                  - Type stored in packed_instace.
+	    loaded_instance       - Buffer with loaded data.
+	    out_txt_instance      - Ptr to buffer where to write txt-data.
+	    out_txt_instance_size - Size of out_txt_instance.
+	    produced_bytes        - Number of bytes that would have been written to out_txt_instance if it was large enough.
+
+	Note:
+	    A loaded instance to unpack is required to be in current platform endian, if not DL_ERROR_ENDIAN_ERROR will be returned.
 */
-dl_error_t DL_DLL_EXPORT dl_txt_unpack( dl_ctx_t dl_ctx,                       dl_typeid_t type,
-                                        const unsigned char* packed_instance,  size_t      packed_instance_size,
-                                        char*                out_txt_instance, size_t      out_txt_instance_size,
-                                        size_t*              produced_bytes );
+dl_error_t DL_DLL_EXPORT dl_txt_unpack_loaded( dl_ctx_t dl_ctx,        dl_typeid_t type,
```

Comment:
sorry for the late reply... family and such take time!

what is the reason you need this, if I understand this it can be used to write a unpacked instance to text. I would like to know the reason as this would be the first function that would act upon a non-packed instance and I have, in the past, made a thing about only acting on packed instances in the API to keep the API-size and implementation smaller and easier to maintain.

Reply:
The reason is that as a user of DL I always have "pointer based/loaded/unpacked" data to start with, and I either want to write it as a binary blob or write it as text. With the same reasoning, I want to load text or binary into something I can use. Now when the pointer patch is so fast, I don't mind that extra step. But there is a high overhead of first doing pack, then writing the json.

So the main motivation was that going between text and packed is rarely what I want, except when I write a "file inspector" or "merge tool" which deals with binary files.

To keep API minimal I could remove the old function, but it felt wrong to remove a function which others already use.

Reply:
I'll buy that... in that case the in-pointer should be "const void*" and not "unsigned char*" as this no longer is a "buffer" per se. That should probably be changed for all "data pointers" but that is a question for later.

Reply:
I made the change, but I can't say that I grasp the "buffer or not" reasoning.

To me an instance with offsets and an instance with pointers are the same thing, they are just relative to different things. I wouldn't mind if they had two different typedefs to make it clear in code what is what, but the void vs uint8 part didn't make sense to me.